### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by sept-en**
The purpose of PR is to fix all warnings we currently have.
For some constants it was weird to add an unnecessary `engine` feature check, so I decided to manually turn off the warning by using the following approach:
```rust
#[cfg_attr(not(feature = "engine"), allow(dead_code))]
```

### CODE REVIEW

**Comment by birchmd** [comment]
Still some warnings showing up in the GH action (appears in `Files changed` tab). Please also fix those in this PR

**Comment by sept-en** [comment]
@birchmd thanks! I fixed those as well. This happens because we are using different toolchains for clippy-checks locally in `Makefile` and in GH workflows. Need to address this thing later as well.

File: src/precompiles/bn128.rs [hunk 1]
```diff
@@ -48,8 +48,11 @@ mod consts {
 pub(super) mod addresses {
     use crate::precompiles;
 
+    #[cfg_attr(not(feature = "engine"), allow(dead_code))]
```
Comment (birchmd) [inline src/precompiles/bn128.rs:51]: Like @mfornet mentioned, we really need to clean up this `engine` feature. What value is it adding?

**Review by birchmd — APPROVED** [review]
Thanks for this!

File: src/precompiles/bn128.rs (hunk 1 above)
Comment (sept-en) [reply src/precompiles/bn128.rs:51]: Yes, I think that this initially appeared for some good purposes, but not it's quite confusing, as now we have 74 usages of it in our code. And this is even after I minimized this a little bit with having one statement for several includes in the file. Let's address this in future PRs though.

**Review by artob — APPROVED** [review]

File: src/precompiles/bn128.rs (hunk 1 above)
Comment (birchmd) [reply src/precompiles/bn128.rs:51]: #154 removes the engine flag

**Pull request merged**